### PR TITLE
cinemageddon: support imdbid

### PR DESCRIPTION
**Pull request by janot**
#4859
fix categories
tried to support dl/ul bonuses, but processing all of them correctly probably needs C#

### CODE REVIEW

**Comment by garfield69** [comment]
none of the team members have an account on this site so:
can you dump the html from the https://cinemageddon.net/help/newuser.php and http://cinemageddon.net/cocks/showpage.php?id=368 and 
also one of a search page with some results showing some of these coins and whatnot?
I can take a look and see what can be done in yaml.

**Comment by janot** [comment]
I've pasted parts of the rules about bonuses here: https://pastebin.com/zDKEzN32 If you really need html of rules pages, I can do that of course

Here's html of the search page where you can see bumpy and coin bonuses combined:
https://pastebin.com/3iPLETAV
And another one with more examples and featured torrents: https://pastebin.com/VGV5idqf

**Comment by RoloSoze** [comment]
@janot Thank you for doing this. If you're also a member of TV Vault, Cinemageddon's sister site, would you be able to implement the IMDb ID search for it as well? That indexer is in C#.

**Comment by janot** [comment]
@RoloSoze Nope, not a member currently.

**Comment by garfield69** [comment]
does the site have anything that alters the upload bonus?

I may have a way to use yaml for the download bumps/coins, I'll try to come up with some test code for you to try out when I next have some free time.

**Comment by janot** [comment]
@garfield69 if I understand correctly, all kinds of bonuses always alter both upload and download stats:

> ...it means the torrent is eligible for a bonus, which takes the amount you **upload/download** on a torrent and **increases/decreases** it by a certain percentage...

Also after re-reading rules about bumpy bonuses I think their minimum possible value is 12%:

> Basically, any torrent that's been unseeded and unloved **for more than a year** can be bumped.....
> ....A bumped torrent will get a bonus of **1% for every month** it's been dead...

Also oldest torrent on site was uploaded at 2007-04-19 which is currently 156 months ago.

Nice, I'd like to see what's possible with yaml here.

**Comment by garfield69** [comment]
gees, complicated much or what?
so if a torrent is bumped to 12%, we are going to get 12% off the download cost, and get 12% added to the upload too?
what is the indicator for the 50%, you said it was `featured` but I'm not sure i'm seeing that flag.

id the golden1.gif the freeleech? so is that 0 cost to download but upload is counted?

**Comment by janot** [comment]
@garfield69 

> 12% off the download cost, and get 12% added to the upload

Yep, I think it works this way.

Featured torrents (50%) have no indicators when searching for torrents. They can only be found in a table above other torrents on e.g. http://cinemageddon.net/browse.php page:
Capture
Capture


Nope, freeleech is this one:  
golden10 `100% bonus` It's not in source code of the pages I provided you. There's currently only 1 freeleech torrent on site.
> so is that 0 cost to download but upload is counted?

I think it's 0% download, 200% upload

other coins:
golden2 `20% bonus`
golden3 `30% bonus`
golden4 `40% bonus`
bumped I suppose you already found bumpy =)

**Comment by garfield69** [comment]
since the support required to enable the bump processing will not be possible until #8504 is implemented, and since it could be a while until devs provide that core support, I am going to add the processing of only the coins to this PR and merge it so your imdbid search is published.

**Comment by garfield69** [comment]
@janot  can you test this and check it still works?

File: src/Jackett.Common/Definitions/cinemageddon.yml [hunk 1]
```diff
@@ -48,61 +48,73 @@
...
-        filters:
-          - name: regexp
-            args: (\d{4}-\d{2}-\d{2})
-      # processing all kinds of up/dl bonus probably needs C#
-      # types of bonuses:
-      # - 4 types of coins (20,30,40,100%) (there's also black coin, but I don't think it's being used) https://cinemageddon.net/help/newuser.php
-      # - featured torrents (50%) on browse page
-      # - bumpy bonus (1%+) http://cinemageddon.net/cocks/showpage.php?id=368 - problematic one,
-      # since it can be applied to torrents already having other bonuses and only the highest one is counted
+        case:
```
Comment (janot) [inline src/Jackett.Common/Definitions/cinemageddon.yml:103]: missing `downloadvolumefactor:`

**Comment by janot** [comment]
@garfield69 everything else seems working

Found a title where cinemageddon's imdb search returns nothing, but jackett returns proper results, lol

**Comment by garfield69** [comment]
Jackett 0.16.310

**Pull request merged**